### ISSUE TITLE
Service crashing after an hour

### ISSUE THREAD

**Issue by davesteele** [davesteele/cloudprint-service#34]
Service is crashing after an hour.

        Jul  9 22:42:49 debian-testing cloudprintd[8392]: Updated Printer MFCJ5910DW
        Jul  9 22:42:50 debian-testing cloudprintd[8392]: Updated Printer MFC490CW
        Jul  9 22:42:50 debian-testing cloudprintd[8392]: Polling for jobs on MFC490CW
        Jul  9 22:42:51 debian-testing cloudprintd[8392]: Polling for jobs on MFCJ5910DW
        Jul  9 22:42:51 debian-testing cloudprintd[8392]: Establishing connection to xmpp server talk.google.com:5223
        Jul  9 22:42:51 debian-testing cloudprintd[8392]: xmpp connection established
        Jul  9 23:42:51 debian-testing cloudprintd[8392]: Traceback (most recent call last):
        Jul  9 23:42:51 debian-testing cloudprintd[8392]: File "/usr/sbin/cloudprintd", line 9, in <module>
        Jul  9 23:42:51 debian-testing cloudprintd[8392]: load_entry_point('cloudprint==0.12', 'console_scripts', 'cloudprint-cmd')()
        Jul  9 23:42:51 debian-testing cloudprintd[8392]: File "/usr/share/cloudprint/cloudprint/cloudprint.py", line 500, in main
        Jul  9 23:42:51 debian-testing cloudprintd[8392]: process_jobs(cups_connection, cpp)
        Jul  9 23:42:51 debian-testing cloudprintd[8392]: File "/usr/share/cloudprint/cloudprint/cloudprint.py", line 400, in process_jobs
        Jul  9 23:42:51 debian-testing cloudprintd[8392]: printers = cpp.get_printers()
        Jul  9 23:42:51 debian-testing cloudprintd[8392]: File "/usr/share/cloudprint/cloudprint/cloudprint.py", line 208, in get_printers
        Jul  9 23:42:51 debian-testing cloudprintd[8392]: 'proxy': self.auth.guid,
        Jul  9 23:42:51 debian-testing cloudprintd[8392]: File "/usr/local/lib/python2.7/dist-packages/requests/models.py", line 819, in json
        Jul  9 23:42:51 debian-testing cloudprintd[8392]: return json.loads(self.text, **kwargs)
        Jul  9 23:42:51 debian-testing cloudprintd[8392]: File "/usr/local/lib/python2.7/dist-packages/simplejson/__init__.py", line 505, in loads
        Jul  9 23:42:51 debian-testing cloudprintd[8392]: return _default_decoder.decode(s)
        Jul  9 23:42:51 debian-testing cloudprintd[8392]: File "/usr/local/lib/python2.7/dist-packages/simplejson/decoder.py", line 370, in decode
        Jul  9 23:42:51 debian-testing cloudprintd[8392]: obj, end = self.raw_decode(s)
        Jul  9 23:42:51 debian-testing cloudprintd[8392]: File "/usr/local/lib/python2.7/dist-packages/simplejson/decoder.py", line 400, in raw_decode  
        Jul  9 23:42:51 debian-testing cloudprintd[8392]: return self.scan_once(s, idx=_w(s, idx).end())
        Jul  9 23:42:51 debian-testing cloudprintd[8392]: simplejson.scanner.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

I just re-establish authentication to see if that helps.

**Comment by davesteele** [comment]
It happened again, after a fresh set of authentication tokens were established.

**Comment by sylvandb** [comment]
You need to change get_printers() to catch the error and show the string. It looks like it might be an empty string, but no way to know. A change something like this (might need generic except instead of except ValueError):

```
     def get_printers(self):
-        printers = self.auth.session.post(
+        r = self.auth.session.post(
             PRINT_CLOUD_URL + 'list',
             {
                 'output': 'json',
                 'proxy': self.auth.guid,
             },
-        ).json()
-        return [PrinterProxy(self, p['id'], p['name']) for p in printers['printers']]
+        )
+        try:
+            printers = r.json()
+        except ValueError:
+            pinfo = []
+            LOGGER.error("get_printers %s, bad json: %s, %s",
+                time.strftime('%Y%m%d-%H%M%S'), r, r.content)
+        else:
+            pinfo = printers.get('printers', [])
+            if self.verbose:
+                LOGGER.debug("get_printers: good json: %s, %s", r, r.content)
+                LOGGER.info("get_printers, %d@%s: %s",
+                    len(pinfo),
+                    time.strftime('%Y%m%d-%H%M%S'),
+                    ', '.join(p['name'] for p in pinfo))
+        return [PrinterProxy(self, p['id'], p['name']) for p in pinfo]
```

**Comment by davesteele** [comment]
It was an authentication timeout, which is fixed in upstream 0.13. Updating, and closing.
